core/templates/domain/topic/editable-topic-backend-api.service.ts

```diff
@@ -16,180 +16,194 @@
         if (successCallback) {
-          // Here also, a dict with 2 fields are passed instead of just 2
-          // parameters, due to the same reason as written for _fetchTopic().
+        // Here also, a dict with 2 fields are passed instead of just 2
+        // parameters, due to the same reason as written for _fetchTopic().
           successCallback({
-            topicDict: angular.copy(response.data.topic_dict),
-            skillIdToDescriptionDict: angular.copy(
-              response.data.skill_id_to_description_dict),
-            skillIdToRubricsDict: angular.copy(
-              response.data.skill_id_to_rubrics_dict)
+            topicDict: cloneDeep(response.topic_dict),
+            skillIdToDescriptionDict: cloneDeep(
+              response.skill_id_to_description_dict),
+            skillIdToRubricsDict: cloneDeep(response.skill_id_to_rubrics_dict)
           });
         }
-      }, function(errorResponse) {
+      }, (errorResponse) => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(errorResponse.error);
         }
       });
-    };
+  }
 
-    return {
-      fetchTopic: function(topicId) {
-        return $q(function(resolve, reject) {
-          _fetchTopic(topicId, resolve, reject);
-        });
-      },
+  fetchTopic(topicId: string): Promise<object> {
+    return new Promise((resolve, reject) => {
+      this._fetchTopic(topicId, resolve, reject);
+    });
+  }
 
-      fetchStories: function(topicId) {
-        return $q(function(resolve, reject) {
-          _fetchStories(topicId, resolve, reject);
-        });
-      },
+  fetchStories(topicId: string): Promise<object> {
+    return new Promise((resolve, reject) => {
+      this._fetchStories(topicId, resolve, reject);
+    });
+  }
 
-      fetchSubtopicPage: function(topicId, subtopicId) {
-        return $q(function(resolve, reject) {
-          _fetchSubtopicPage(topicId, subtopicId, resolve, reject);
-        });
-      },
-
-      /**
-       * Updates a topic in the backend with the provided topic ID.
-       * The changes only apply to the topic of the given version and the
-       * request to update the topic will fail if the provided topic
-       * version is older than the current version stored in the backend. Both
-       * the changes and the message to associate with those changes are used
-       * to commit a change to the topic. The new topic is passed to
-       * the success callback, if one is provided to the returned promise
-       * object. Errors are passed to the error callback, if one is provided.
-       */
-      updateTopic: function(
-          topicId, topicVersion, commitMessage, changeList) {
-        return $q(function(resolve, reject) {
-          _updateTopic(
-            topicId, topicVersion, commitMessage, changeList,
-            resolve, reject);
-        });
-      },
+  fetchSubtopicPage(topicId: string, subtopicId: number): Promise<object> {
```

Comment:
Promise => Promise. (Notice the capital O)

Reply:
Done